lib/fs.js

```diff
@@ -1353,7 +1373,10 @@ function FSWatcher() {
   this._handle.onchange = function(status, event, filename) {
     if (status < 0) {
       self._handle.close();
-      const error = errnoException(status, `watch ${filename}`);
+      const error = !filename ?
+          errnoException(status, 'Error watching file for changes:') :
+          errnoException(status,
+                         `Error watching file ${filename} for changes:`);
```

Comment:
Well, it's unrelated to the Buffer / encoding change but given that the current error message is fairly useless when filename is not returned, it certainly seems worthwhile since I've got my hands in this code anyway.